added ability to override build scripts used

kevinsperrine opened the pull request:
I use the typescript build scripts with CRA along with react-app-rewired to make some overrides without ejecting. This allows adding a custom variable to serverless.yml and build with those configs. Not fully tested. Feel free to change or ignore.

```
# serverless.yml
custom:
  build-cra:
    command: react-app-rewired
    params:
      - --scripts-version
      - react-scripts-ts
```

kevinsperrine committed:
added ability to override build scripts used

I use the typescript build scripts with CRA along with react-app-rewired to make some overrides without ejecting. This allows adding a custom variable to serverless.yml and build with those configs. Not fully tested. Feel free to change or ignore.

```
# serverless.yml
custom:
  build-cra:
    command: react-app-rewired
    params:
      - --scripts-version
      - react-scripts-ts
```

lyuzashi commented:
Hi @kevinsperrine thank you for the contribution. Sorry it's taken me so long to have a look at this.

It's also been a while since I've used CRA or Serverless, so I've forgotten how this plugin works 😂

Allowing for a custom build command is pretty cool! Would you be able to share a project example that I can test? 

Would also be great to add your serverless.yml snippet to the readme in your PR.

Will be happy to merge and release once I can try it out.

Thanks again!

lyuzashi commented on index.js:
```diff
@@ -33,12 +33,21 @@ class BuildReactForSyncPlugin {
       const env = { ...process.env, ...outputs };
-      const build = require.resolve('react-scripts/scripts/build');
-      const result = spawn.sync( 'node', [build], { stdio: 'inherit', env });
+      const buildopts = this.serverless.service.custom['build-cra'];
```
For consistency's sake, could we call the custom options "build-create-react-app"?

kevinsperrine committed:
change custom array name to build-create-react-app

kevinsperrine committed:
include details on custom build configuration usage in readme

kevinsperrine committed:
check to make sure params exist

kevinsperrine commented:
@lyuzashi, I've changed the custom config name, included some configuration details in the readme, added a fix to ensure the params exist, and create a minimal-repo for you to see it in action. The minimal repo is pointing at my patch-1 branch. 

https://github.com/kevinsperrine/serverless-plugin-create-react-app-test

lyuzashi commented:
Thanks @kevinsperrine for updating everything as I requested so quickly! The example repository was super handy testing this.

I noticed the plugin fails if there are no custom options set

> `Cannot read property 'build-create-react-app' of null`

If you could also include a guard against this, then I'm happy to merge.

lyuzashi commented on index.js:
```diff
@@ -33,12 +33,23 @@ class BuildReactForSyncPlugin {
       const env = { ...process.env, ...outputs };
-      const build = require.resolve('react-scripts/scripts/build');
-      const result = spawn.sync( 'node', [build], { stdio: 'inherit', env });
+      const buildopts = this.serverless.service.custom['build-create-react-app'];
```
Can we guard against no options set with something like
```js
const buildopts = this.serverless.service.custom && this.serverless.service.custom['build-create-react-app'];
```

kevinsperrine committed:
check that custom exists

kevinsperrine commented:
@lyuzashi Absolutely. Good catch. I added it in.

lyuzashi commented:
Thank you again for your contribution @kevinsperrine! Merged and published 1.1.0 on NPM. Hope it's useful for you!

kevinsperrine commented:
@lyuzashi Thank you! It definitely will be on my current project.
